### ISSUE TITLE
bug report: sumatrapdf crashed

### ISSUE THREAD

**Issue by shlomiag** [sumatrapdfreader/sumatrapdf#2131]
3.4.14028 pre-release 64-bit

[Please see here for `sumatrapdfcrash.txt` with private info removed.](https://gist.github.com/shlomiag/0b064752ddbfc99983ddba86eddfbf65)

For the `sumatrapdfcrash.dmp` file, please contact me via private communications (could not remove private info from binary file).

Thanks,

Shlomi

**Comment by GitHubRulesOK** [comment]
wow thats a lot of files with the same name :-), however it appears at first glance a pdf file was incomplete ? and with lots of LaTeX activity so were you compiling at the time ?

**Comment by shlomiag** [comment]
No, they're actually many different files. But I renamed them all, for
privacy...  :-)

Yes -- I'm using Latex a lot, recompiling files frequently, with sumtrapdf
reloading them in the background. That's my TeX work setup...  :-)

**Comment by GitHubRulesOK** [comment]
@kjk 
that makes me question, that since MuPDF is "effective" at correcting PDF files in progress how to guard against those that are undergoing external changes?

Is there a means to ignore "currently open" pdf file during "compilation preview" also there are those changes that work without a synctex call in progress via auxillary TeX (or other) commands.

**Comment by shlomiag** [comment]
@GitHubRulesOK , @kjk , I'd like to note that for me, sumatrapdf is ideal for TeX work _because_ it is able to handle frequent PDF re-compiles by an external application. That's actually the reason I chose it over quite a few others.   ;-)

**Comment by kjk** [comment]
Looks like mupdf got into a state that caused it to crash when trying to free a document.

Do you still have the file that caused the crash? Without access to a reproducible case I can't do much.

**Comment by shlomiag** [comment]
The crash doesn't seem to be reproducible. It's a file I've been editing for a long while now (sorry, unable to share this one...), and have continued editing also after the crash. I've never seen sumatra crash like that before. It seems as if a particular timing of window-switching from TeX to sumatra caught the latter in surprise, leading to a crash.

Would the _sumatrapdfcrash.dmp_ be of use?

**Comment by GitHubRulesOK** [comment]
@kjk 
Whilst a TeX user is making incremental edits the PDF can often be caught in an unstable state due to bad edit (say wrong image name or path) the causes are infinite due to multiple small packages dependant on the prior one not failing. Its not helped that during compilation a new command may require a package update from an online source so can stutter for a few commands. With luck the PDF is xrefed correctly during a refresh but timing can be upset by a second command holding the file part built as its own.

**Comment by kjk** [comment]
@shlomiag sumatrapdfcrash.dmp won't help much. I have a pretty good idea of which code crashed but the cause of crash is not obvious so I really need to be able to reproduce this under the debugger etc.

**Comment by GitHubRulesOK** [comment]
@shlomiag you should be aware you cannot rebuild a broken .tex from a .pdf thus backup your valuable irreplacable source.tex SumatraPDF should not be a problem but TeX Works could, the pdf can be easily replaced.
Out of interest is it Static Live TeX or MiKTeX and if the latter is it set to update on the fly?, and again is it possible you added a new \package or \usage before the fail?

**Comment by shlomiag** [comment]
@kjk, the crash caught me in the middle of an editing session. When doing so, I save and recompile very often, and so don't have anymore the exact PDF which triggered the crash. I tried scrolling back through the editing history (Ctrl+Z) and recompiling on suspicious points, but nothing re-triggered the crash so far. Could it help if I send you a similar PDF from a broken compilation?

@GitHubRulesOK, I'm not sure that I follow. I'm using TeXstudio on top of MiKTex. But I added no package recently. I think that the crash happened on a miscompiled $paritial_\beta$ command (should be $partial_\beta$), but I'm not sure.

**Comment by kjk** [comment]
@shlomiag Only the file that causes a crash would be of help. If it ever happens again, try preserving the file. If you get a consistent crash, e-mail me the file at kkowalczyk@gmail.com

**Comment by shlomiag** [comment]
I see. Will do, next time I get a crash.

Thank you both for your support!    :-))

On Aug 19, 2021, 12:37 AM, at 12:37 AM, Krzysztof Kowalczyk ***@***.***> wrote:
***@***.*** Only the file that causes a crash would be of help. If it

**Comment by kjk** [comment]
Can't make progress without being able to reproduce.
